### PR TITLE
[New] Suspicious Microsoft 365 Mail Access by ClientAppId

### PR DESCRIPTION
**Pull request by Samirbous**
Using New Term rule type identifies when a Microsoft 365 Mailbox is accessed by a ClientAppId that was observed for the fist time during the last 10 days.

https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-193a https://www.elastic.co/guide/en/beats/filebeat/current/exported-fields-o365.html

KQL validation : 

image

### CODE REVIEW

**Review by Aegrah — APPROVED** [review]
LGTM. Might be good to bump down to low severity given that employees go on holidays, they leave/join the company etc. But that's up to you.

**Review by Mikaayenson — APPROVED** [review]

File: rules/integrations/o365/initial_access_microsoft_365_abnormal_clientappid.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+license = "Elastic License v2"
+name = "Suspicious Microsoft 365 Mail Access by ClientAppId"
+note = """## Setup
+
+The Office 365 Logs Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.
+"""
+references = ["https://www.cisa.gov/news-events/cybersecurity-advisories/aa23-193a"]
+risk_score = 47
+rule_id = "48819484-9826-4083-9eba-1da74cd0eaf2"
+severity = "medium"
```
Comment (Mikaayenson) [inline rules/integrations/o365/initial_access_microsoft_365_abnormal_clientappid.toml:28]: should we bump this?

Comment (Samirbous) [reply rules/integrations/o365/initial_access_microsoft_365_abnormal_clientappid.toml:28]: can be bumped up based on feedback/noise level

**Comment by Samirbous** [comment]
> oyees go on holidays, they leave/join

that's possible users can bump up interval to 20days to reduce noise from this scenario

**Pull request merged**